controllers/snapshot/snapshot_adapter.go

```diff
@@ -217,8 +219,18 @@ func (a *Adapter) EnsureStaticIntegrationPipelineRunsExist() (controller.Operati
 			}
 		}()
 
+		var errsForPLRCreation error
 		for _, integrationTestScenario := range *integrationTestScenarios {
 			integrationTestScenario := integrationTestScenario //G601
+			scenarioStatusCondition := meta.FindStatusCondition(integrationTestScenario.Status.Conditions, gitops.IntegrationTestScenarioValid)
+			if scenarioStatusCondition.Status == metav1.ConditionFalse {
```

Comment:
Maybe we could have a new function that checks if the integrationTestScenario is valid and returns a boolean? Something like `IsScenarioValid(scenario)`?

Reply:
I thought to add that function but because I need to call `scenarioStatusCondition := meta.FindStatusCondition(integrationTestScenario.Status.Conditions, gitops.IntegrationTestScenarioValid)` to write `statusCondition.Message` to snapshot annotation.
That function is added now, thanks.